### ISSUE TITLE
Undefined symbols for architecture x86_64

### ISSUE THREAD

**Issue by jimUpdate** [CocoaLumberjack/CocoaLumberjack#930]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [ ] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | Xcode 9
Swift Version | 3.0
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Swift3.0 xcode9

```
Undefined symbols for architecture x86_64:
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtFfA0_", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtFfA1_", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtFfA5_", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtFfA6_", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtFfA7_", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
  "__T015CocoaLumberjack10DDLogDebugySSyXK_SC0C5LevelO5levelSi7contexts12StaticStringV4fileAH8functionSu4lineypSg3tagSb12asynchronousSo0C0C5ddlogtF", referenced from:
      __T07Xinxike8LogUtilsC3logySQySSG3msg_tFZ in LogUtils.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

LogUtils 代码
```swift
import CocoaLumberjack

class LogUtils{
    class func log(msg:String!){
        
        #if DEBUG
//            NSLog(msg)
//            LogUtilsOC.log(msg)
            DDLogDebug(msg)
        #endif
    }
    
    
    class func initLogUtils(){
        
        DDLog.add(DDTTYLogger.sharedInstance) // TTY = Xcode console
        DDLog.add(DDASLLogger.sharedInstance) // ASL = Apple System Logs
        
        let fileLogger: DDFileLogger = DDFileLogger() // File Logger
        fileLogger.rollingFrequency = TimeInterval(60*60*24)  // 24 hours
        fileLogger.logFileManager.maximumNumberOfLogFiles = 7
        DDLog.add(fileLogger)
        
    }
    
}
```

podfile 代码
```ruby
platform :ios, ‘9.0’
use_frameworks!
inhibit_all_warnings!#忽略警告

target 'Demo' do
    pod 'Moya/RxSwift', '8.0.5'
    pod 'RxSwift', '3.6.1'
    pod 'RxCocoa', '3.6.1'
    pod 'SwiftyJSON','3.1.4'
    pod 'Toaster','2.0.4'#吐司
    pod 'SnapKit', '3.2.0'#代码添加约束
    pod 'MJRefresh','3.1.15'
    pod 'HGPlaceholders','0.2.3'
    pod 'Kingfisher', '4.3.0'#加载图片
    pod 'HTMLString', '3.0.0'#富文本处理工具
    pod 'JRDB','2.1.1'#数据库
    pod 'SwipeMenuViewController','2.0.2' #tab页切换
    pod 'FSCalendar', '2.7.9'#值班日历
    pod 'SwiftEventBus','2.1.2'
    pod 'SVProgressHUD','2.2.5'#弹窗
    pod 'NextGrowingTextView','1.1.0'#多文本输入
    pod 'SLExpandableTableView','1.3.1'#可折叠的UitableView
    pod 'TZImagePickerController','1.9.6'#图片选择框架
    pod 'SearchEmojiOnString', '~> 1.0'#表情校验[不固定版本,因为需要包含尽可能多的表情]
    pod 'IQKeyboardManagerSwift','5.0.8'
    pod 'CocoaLumberjack/Swift'#Log收集器用于平时debug
    
end

target :WordPress do
    pod 'CocoaLumberjack'
    pod 'NSObject-SafeExpectations','0.0.2'
end


post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['SWIFT_VERSION'] = '3.0'
config.build_settings['ARCHS'] = 'i386 x86_64 armv7s armv7 arm64'
        end
    end
end
```

DDLogDebug(msg) 报错

**Comment by bpoplauschi** [comment]
@jimUpdate thanks for your interest in CocoaLumberjack.
I've re-added the issue template info to the description, as this information is vital to us.

If you are using the latest version of CocoaLumberjack, please note this version requires Swift 4 (see [our Readme](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/README.md#requirements)).

For Swift 3.0, use the 3.2 version as described in the [Backwards compatibility section](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/README.md#backwards-compability).

**Comment by Sopan** [comment]
@bpoplauschi Looks like there is a new architecture added "arm64e" on new iPhone hardware (iPhone Xs and iPhone Xs Max) because of which I am seeing an issue (undefined architecture arm64e)  whenever I try building it my app on these devices through Xcode 10.

The warning is
```
ignoring fie ./Lumberjack/CocoaLumberjack.frameowrk/CocoaLumberjack, missing required architecture arm64e in file ./Lumberjack/CocoaLumberjack.frameowrk/CocoaLumberjack (4 slices)
```
and corresponding error

```
"_OBJC_CLASS_$_DDFileLogger", referenced from:
  "_OBJC_CLASS_$_DDLog", referenced from:
  "_OBJC_CLASS_$_DDLogFileManagerDefault", referenced from:
  "_OBJC_CLASS_$_DDTTYLogger", referenced from:
  "_OBJC_IVAR_$_DDLogMessage._flag", referenced from:
  "_OBJC_IVAR_$_DDLogMessage._message", referenced from:
  "_OBJC_IVAR_$_DDLogMessage._timestamp", referenced from:
  "_OBJC_METACLASS_$_DDLogFileManagerDefault", referenced from:
```

Looks like currently the framework built for following architectures
```i386 x86_64 armv7 arm64```
I used ```xcrun -sdk iphoneos lipo -info ./CocoaLumberjack``` to figure it out.

Do you have plans to support this new architecture in cocoalumberjack.

Also, would you like me to create a separate issue for it?

**Comment by bpoplauschi** [comment]
@Sopan we need to look into how we compute that list of architectures. 
I thought leaving the setting (`VALID_ARCHS`) unfilled will use the default value which should include all the needed architectures.
I suggest you create a separate issue to track this. Also, feel free to look into creating a PR.

Thanks for reporting.

**Comment by bpoplauschi** [comment]
Closing this issue, the new conversation moved to #966
